test/contracts/acl/acl_params.js

```diff
@@ -0,0 +1,101 @@
+  })
+
+  context('> ACL Oracle', () => {
+    describe('when the oracle accepts', () => {
+      let acceptOracle, param
+
+      before(async () => {
+        acceptOracle = await AcceptOracle.new()
+        param = paramForOracle(acceptOracle.address)
+      })
+
+      describe('when permission is set for ANY_ADDR', () => {
+        it('ACL allows actions', async () => {
+          await acl.grantPermissionP(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isTrue(await acl.hasPermission(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE))
+        })
+      })
+
+      describe('when permission is set for specific address', async () => {
+        it('ACL allows actions when permisison is set for a specific address', async () => {
+          await acl.grantPermissionP(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isTrue(await acl.hasPermission(permissionsRoot, mockAppAddress, MOCK_APP_ROLE))
+        })
+      })
+    })
+
+    describe('when the oracle modifies state', () => {
+      let stateModifyingOracle, param
+
+      before(async () => {
+        stateModifyingOracle = await StateModifyingOracle.new()
+        param = paramForOracle(stateModifyingOracle.address)
+      })
+
+      describe('when permission is set for ANY_ADDR', () => {
+        it('ACL disallows actions', async () => {
+          await acl.grantPermissionP(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isFalse(await acl.hasPermission(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE))
+        })
+      })
+
+      describe('when permission is set for specific address', async () => {
+        it('ACL disallows actions', async () => {
+          await acl.grantPermissionP(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isFalse(await acl.hasPermission(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE))
+        })
+      })
+    })
+
+    describe('when the oracle uses all available gas', () => {
+      let overGasLimitOracle, param
+
+      before(async () => {
+        overGasLimitOracle = await OverGasLimitOracle.new()
+        param = paramForOracle(overGasLimitOracle.address)
+      })
+
+      describe('when permission is set for ANY_ADDR', () => {
+        // Note `evalParams()` is called twice when calling `hasPermission` for `ANY_ADDR`
+        it('ACL disallows actions', async () => {
+          await acl.grantPermissionP(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isFalse(await acl.hasPermission(ANY_ADDR, mockAppAddress, MOCK_APP_ROLE))
+        })
+      })
+
+      describe('when permission is set for specific address', async () => {
+        // Note `evalParams()` is only called once when calling `hasPermission` for a specific address
+        it('ACL disallows actions', async () => {
+          await acl.grantPermissionP(permissionsRoot, mockAppAddress, MOCK_APP_ROLE, [param])
+          assert.isFalse(await acl.hasPermission(permissionsRoot, mockAppAddress, MOCK_APP_ROLE))
```

Comment:
Note that in these cases that are now not throwing due to sending all gas (truffle by default is using 6.9M gas), it may be interesting to experiment with sending a transaction with lower gas values to see if they start failing and adding some tests around this.